compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -269,8 +283,17 @@ class InstrumentCoverage extends MacroTransform with IdentityDenotTransformer:
             transformDefDef(tree)
 
           case tree: PackageDef =>
-            // only transform the statements of the package
-            cpy.PackageDef(tree)(tree.pid, transform(tree.stats))
+            if (isFileIncluded(tree.srcPos.sourcePos.source) && isClassIncluded(tree.symbol))
+              // only transform the statements of the package
+              cpy.PackageDef(tree)(tree.pid, transform(tree.stats))
+            else
+              tree
+
+          case tree: TypeDef =>
+            if (isFileIncluded(tree.srcPos.sourcePos.source) && isClassIncluded(tree.symbol))
```

Comment:
This does not match the description of the setting. It seems we apply the regexes to class names, whereas the doc says it only applies to packages.

Reply:
I tried replicating the behaviour of Scala 2 scoverage filters as closely as possible. The package filter in Scala 2 is also applied to classes and modules. I think that the logic should stay the same, we can change the name and doc instead.
I'll change the doc to `List of regexes for packages, classes and modules to exclude from coverage.`
Do you have a suggestion for a better option name than `-coverage-exclude-packages`?
Maybe something like:
- `-coverage-exclude-classlikes` (classlike is used in scaladoc)
- `-coverage-exclude-by-name` or `-coverage-exclude-by-fqn`
- `-coverage-exclude-typedefs`

Reply:
If Scaladoc uses "classlike", I guess we can use the same, yes.